### PR TITLE
Add serialization support for XmlDocument and XmlElement in XmlSerializer

### PR DESCRIPTION
**Pull request by khdang**
Add back the code that was removed when moving serialization to .NET Core. Issue #1256.

### CODE REVIEW

**Comment by khdang** [comment]
@dotnet-bot test this please

File: src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs [hunk 1]
```diff
@@ -913,7 +914,7 @@ public class SimpleBase
     [KnownType(typeof(SimpleBaseDerived2))]
     public class GenericBase2<T, K>
         where T : new()
-        where K : new()
+    where K : new()
```
Comment (stephentoub) [inline src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs]: Nit: why this spacing change?

**Comment by stephentoub** [comment]
As a replacement for #1343, LGTM.

File: src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs (hunk 1 above)
Comment (khdang) [inline src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs]: This is unintentional. I've reverted this change and also done a CodeFormatter pass through serialization code.

**Comment by stephentoub** [comment]
Formatting changes look fine.

**Pull request merged**